### ISSUE TITLE
[BUG] Consume proxy from environment variables in Firefox

### ISSUE THREAD

**Issue by mxschmitt** [microsoft/playwright#6094]
**Context:**
- Playwright Version: 1.10... probably all
- Operating System: Linux
- Node.js version: 15
- Browser: Firefox

**Code Snippet**

Any `firefox.launch()` independent from the language.

This works:

```js
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    browser = p.firefox.launch(proxy={
      "server": "http://squid:3128"
    })
    page = browser.new_page()
    page.goto("https://www.example.com")
    browser.close()
```

Without manually specifying the proxy it does not work and I get `NS_ERROR_CONNECTION_REFUSED` in `page.goto`. The env vars look the follows:

```
HTTPS_PROXY=http://squid:3128
http_proxy=http://squid:3128
```

**Describe the bug**

Chromium and WebKit pick the default proxy configuration up from the environment variables, this is useful for corporate environments which then directly work by default when their sys-admins configured it correctly instead of that the user has to manually set it in the launch options.

-> Use `http_proxy` / `HTTPS_PROXY` from the env vars.

Code pointer to Chromium how they handle it: https://source.chromium.org/chromium/chromium/src/+/master:net/proxy_resolution/proxy_config_service_linux.cc;l=171?q=https_proxy

**Comment by JoelEinbinder** [comment]
Is this implementable by just making the proxy server default to the HTTPS_PROXY option in the frontend? Or are there more environment variables and options we would need to support?

**Comment by mxschmitt** [comment]
Not sure if `http_proxy` or `HTTPS_PROXY` should be used there. Since browsers make it depending on the protocol and PW not.

Seems like Firefox normally does it already internally but the Playwright patch might broke it.

**Comment by mxschmitt** [comment]
I did some more investigation. I found in my use-case two bugs or issues:

a) I had no protocol specified in the `http_proxy` environment variable, Firefox does not like that and ignores it then.
b) There is a _bug_ in the Firefox implementation, that it only checks `https_proxy` lowercase, but it should actually check it upper-case, thats how cURL is doing it. See here: https://github.com/curl/curl/blob/3085ccfae996bb0fa606d2c7bc6783dc15d76a30/docs/MANUAL.md#environment-variables

By adding the protocol and using the env var in lower-case it works as expected.

So definitely not a Playwright bug, both issues could be fixed upstream and the second one is definitely a weird logic where its not following the standard (actually there is no proxy env var standard etc.). Let me know what you think, then I can file the bug/s upstream.
